dds_web/api/project.py

```diff
@@ -767,19 +767,26 @@ def delete_project_contents(project, delete_bucket=False):
             sqlalchemy.exc.OperationalError,
             AttributeError,
         ) as sqlerr:
-            raise DeletionError(
-                project=project.public_id,
-                message=str(sqlerr),
-                alt_message=(
+            if flask.request:
+                raise DeletionError(
+                    project=project.public_id,
+                    message=str(sqlerr),
+                    alt_message=(
+                        "Project bucket contents were deleted, but they were not deleted from the "
+                        "database. Please contact SciLifeLab Data Centre."
+                        + (
+                            "Database malfunction."
+                            if isinstance(sqlerr, sqlalchemy.exc.OperationalError)
+                            else "."
+                        )
+                    ),
+                ) from sqlerr
+            else:
+                error_msg = (
```

Comment:
To avoid having the err_message twice in the code and to have the same text logged:

```suggestion
            error_msg = (
                "Project bucket contents were deleted, but they were not deleted from the "
                "database. Please contact SciLifeLab Data Centre"
                + (
                    ". Database malfunction."
                    if isinstance(sqlerr, sqlalchemy.exc.OperationalError)
                    else "."
            )
            )
            if flask.request:
                raise DeletionError(
                    project=project.public_id,
                    message=str(sqlerr),
                    alt_message=error_msg,
                ) from sqlerr
            else:
                flask.current_app.logger.exception(error_msg)
```